### ISSUE TITLE
[FR] Add `-h` short help option

### ISSUE THREAD

**Issue by abitrolly** [mlflow/mlflow#9117]
### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

The proposal is to add `-h` help shortcut for CLI users.

### Motivation

> #### What is the use case for this feature?

Access help faster. Meet expectations that Python argparse tools have it.

> #### Why is this use case valuable to support for MLflow users in general?

Saves time.

> #### Why is this use case valuable to support for your project(s) or organization?

Same.

> #### Why is it currently difficult to achieve this use case?

Not difficult. https://click.palletsprojects.com/en/8.1.x/documentation/#help-parameter-customization needs to be added here https://github.com/mlflow/mlflow/blob/dc5885a80c06981379d9b5379506c4fc746b5412/mlflow/cli.py#L48

### Details

Right now `-h` is not supported.

```sh
$ mlflow run
Usage: mlflow run [OPTIONS] URI
Try 'mlflow run --help' for help.

Error: No such option: -h
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by abitrolly** [comment]
@serena-ruan that must be `mlflow models serve` and `mlflow server` - https://mlflow.org/docs/latest/cli.html

But it is not necessary to override `-h` with `--help` for these commands, from [`click` docs](https://click.palletsprojects.com/en/8.1.x/documentation/#help-parameter-customization).

> If a command itself implements a parameter with the same name, the default help parameter stops accepting it.

**Comment by serena-ruan** [comment]
Yes that's true, but we want to avoid confusion about using -h as '--host' in some places but as '--help' in other places :) BTW --help doesn't seem that long lol, and it's not quite often people need to type this command right?

**Comment by abitrolly** [comment]
@serena-ruan is there a real confusion here? As a CLI user, I don't feel confused when `-t` option means timeout for `mlflow models serve` and content-type for `mlflow models predict`. It is completely okay for me if `-h` will be host for server commands and help for everything else.

> BTW --help doesn't seem that long lol, and it's not quite often people need to type this command right?

I would be interested to ask other Linux users, but I type `-h` very often, like several times a day. So every 15th command in CLI is a help command. Because it is impossible to remember all options.
```
✗ history | wc -l                 
10285
✗ history | grep " -h" | wc -l   
615
✗ history | grep " --help" | wc -l
140
```
I also use things like https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/git#aliases to save me from typing.

**Comment by harupy** [comment]
We might add `--host` to `mlflow run` in the future. We want to keep `-h` for that.

> I type -h very often, like several times a day

Is `several times a day` "very often"?

**Comment by harupy** [comment]
Is there really an expectation that `-h` is a short version of `--help`?

```
> cat -h
cat: invalid option -- 'h'
Try 'cat --help' for more information.

> touch -h
touch: missing file operand
Try 'touch --help' for more information.

> grep -h
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.

> sed -h
sed: invalid option -- 'h'
```

**Comment by abitrolly** [comment]
> Is several times a day "very often"?

Yes, because it disrupts expected user flow it is a bad https://github.com/mlflow/mlflow/labels/area%2Fuiux. And it can be fixed without degrading the flow for `mlflow` serve users.

> ```
> > cat -h
> cat: invalid option -- 'h'
> Try 'cat --help' for more information.
> 
> > touch -h
> touch: missing file operand
> Try 'touch --help' for more information.
> 
> > grep -h
> Usage: grep [OPTION]... PATTERNS [FILE]...
> Try 'grep --help' for more information.
> 
> > sed -h
> sed: invalid option -- 'h'
> ```

```
$ man mlflow
No manual entry for mlflow
```
> Is there really an expectation that `-h` is a short version of `--help`?

Yes. And my UX hypothesis is that the expectation is common for these types of users 

1. People who were born after `cat`, `sed`, `touch` and `grep` were written. 
2. People who write and use command line tools written with Python.
3. People who use shortcuts.

`meta-llama/Llama-2-70b-chat-hf` also confirms that `-h` is widely used for help https://hf.co/chat/r/Byv_aIr

> Can you add an alias like `mrh` which is an alias for `mlflow run --help`?

I can send a PR to remove [this stuff](https://github.com/mlflow/mlflow/blob/master/CODE_OF_CONDUCT.rst#our-standards) from the repo if you want.

**Comment by harupy** [comment]
Reopened this issue. We'll keep this issue open to collect upvotes/feedback.

**Comment by abitrolly** [comment]
@harupy thanks! :D

**Comment by B-Step62** [comment]
Closing due to inactivity. Feel free to reopen if there is renewed interest in a `-h` short help alias.
